kr8s/_api.py

```diff
@@ -325,21 +325,26 @@ async def async_lookup_kind(self, kind) -> tuple[str, str, bool]:
         if group:
             version = f"{group}/{version}"
         for resource in resources:
+            name = resource["name"]
+            singular_name = resource["singularName"]
+            if len(singular_name) == 0 and len(name) > 1:
+                singular_name = name.lower()[:-1]
```

Comment:
I've never seen an example of a resource where the singular name is not set. AFAIK it is not an optional field.

I'm also concerned about the logic here for dropping the `s` from the plural name. What happens for resources with plurals like `NetworkPolicies`?

Reply:
>with plurals like NetworkPolicies?
thats a good question, I have no answer to it.

Reply:
is it possible to configure k8s cluster to include real singularName instead of just an empty string?
I did not find any references to it in the documentation